source/corvusoft/restbed/detail/socket_impl.cpp

```diff
@@ -159,6 +159,7 @@ namespace restbed
         void SocketImpl::write( const Bytes& data, const function< void ( const error_code&, size_t ) >& callback )
         {
             m_buffer = make_shared< Bytes >( data );
+            if (m_is_open) {
```

Comment:
Or also possible is to let the websocket check for is_open before trying to write or read, an example being the listen fucntion in web_socket_impl.cpp.